packages/server/src/lib/InMemorySessionStore.ts

```diff
@@ -0,0 +1,359 @@
+    const now = Date.now();
+    const expiredIds: string[] = [];
+
+    for (const [sessionId, node] of this.items.entries()) {
+      if (this.ttlMs > 0 && node.expiry <= now) {
+        expiredIds.push(sessionId);
+      }
+    }
+
+    for (const sessionId of expiredIds) {
+      await this.deleteSession(sessionId);
+    }
+  }
+
+  /**
+   * Bump a node to the end of the LRU list (most recently used)
+   */
+  private bumpNode(node: LruNode): void {
+    // Update expiry
+    node.expiry = this.ttlMs > 0 ? Date.now() + this.ttlMs : Infinity;
+
+    if (this.last === node) {
+      return; // Already most recent
+    }
+
+    const { prev, next } = node;
+
+    // Unlink from current position
+    if (prev) prev.next = next;
+    if (next) next.prev = prev;
+    if (this.first === node) this.first = next;
+
+    // Link to end
+    node.prev = this.last;
+    node.next = null;
+    if (this.last) this.last.next = node;
+    this.last = node;
+
+    if (!this.first) this.first = node;
+  }
+
+  /**
+   * Evict the least recently used session
+   */
+  private async evictLru(): Promise<void> {
+    const lruNode = this.first;
+    if (!lruNode) return;
+
+    await this.deleteSession(lruNode.sessionId);
+  }
+
+  async startSession(params: CreateSessionParams): Promise<SessionStartResult> {
+    // Generate session ID or use provided browserbase session ID
+    const sessionId = params.browserbaseSessionID ?? randomUUID();
+
+    // Store the session
+    await this.createSession(sessionId, params);
+
+    return {
+      sessionId,
+      available: true,
+    };
+  }
+
+  async endSession(sessionId: string): Promise<void> {
+    await this.deleteSession(sessionId);
+  }
+
+  async hasSession(sessionId: string): Promise<boolean> {
+    const node = this.items.get(sessionId);
+    if (!node) return false;
+
+    // Check if expired
+    if (this.ttlMs > 0 && node.expiry <= Date.now()) {
+      await this.deleteSession(sessionId);
+      return false;
+    }
+
+    return true;
+  }
+
+  async getOrCreateStagehand(
+    sessionId: string,
+    ctx: RequestContext,
+  ): Promise<V3> {
+    const node = this.items.get(sessionId);
+
+    if (!node) {
+      throw new Error(`Session not found: ${sessionId}`);
+    }
+
+    // Check if expired
+    if (this.ttlMs > 0 && node.expiry <= Date.now()) {
+      await this.deleteSession(sessionId);
+      throw new Error(`Session expired: ${sessionId}`);
+    }
+
+    // Bump to most recently used
+    this.bumpNode(node);
+
+    // Update logger reference for this request
+    if (ctx.logger) {
+      node.loggerRef.current = ctx.logger;
+    }
+
+    // If V3 instance exists, return it
+    if (node.stagehand) {
+      return node.stagehand;
+    }
+
+    // Create V3 instance (lazy initialization)
+    const options = this.buildV3Options(node.params, ctx, node.loggerRef);
+
+    // Import V3 dynamically to avoid circular dependency
+    const { V3: V3Class } = await import("@browserbasehq/stagehand");
```

Comment:
do we still need this now that it's a separate package? can we just do a relative import?